src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -1351,48 +1374,56 @@ class NodeConfig(object):
         # Second loop determines device types and their location
         # under /dev. Only look for block and character devices.
         for path in find_files(os.path.join(os.sep, 'dev'), kind='bc',
                                follow_mounts=False):
             # If the stat fails, log it and continue.
-            try:
-                statinfo = os.stat(path)
-            except KeyboardInterrupt:
-                raise
-            except:
-                pbs.logmsg(pbs.EVENT_DEBUG2, '%s: Stat error on %s' %
-                           (caller_name(), path))
+            devinfo = self._devinfo(path)
+            if not devinfo:
                 continue
-            major = os.major(statinfo.st_rdev)
-            minor = os.minor(statinfo.st_rdev)
-            pbs.logmsg(pbs.EVENT_DEBUG4, 'Path: %s, Major: %d, Minor: %d' %
-                       (path, major, minor))
-            for cls in devices:
-                for inst in devices[cls]:
-                    if 'type' not in devices[cls][inst]:
-                        devices[cls][inst]['type'] = None
-                    if 'device' not in devices[cls][inst]:
-                        devices[cls][inst]['device'] = None
-                    if devices[cls][inst]['major'] == major:
-                        if devices[cls][inst]['minor'] == minor:
-                            if stat.S_ISCHR(statinfo.st_mode):
-                                devices[cls][inst]['type'] = 'c'
-                            else:
-                                devices[cls][inst]['type'] = 'b'
-                            devices[cls][inst]['device'] = path
+
+            for dclass in devices:
+                for inst in devices[dclass]:
+                    if 'type' not in devices[dclass][inst]:
+                        devices[dclass][inst]['type'] = None
+                    if 'device' not in devices[dclass][inst]:
+                        devices[dclass][inst]['device'] = None
+                    if devices[dclass][inst]['major'] == devinfo['major']:
+                        if devices[dclass][inst]['minor'] == devinfo['minor']:
+                            devices[dclass][inst]['type'] = devinfo['type']
+                            devices[dclass][inst]['device'] = path
         # Check to see if there are gpus on the node and copy them
         # into their own dictionary.
         devices['gpu'] = {}
         gpus = self._discover_gpus()
         if gpus:
-            for cls in devices:
-                for inst in devices[cls]:
+            for dclass in devices:
+                for inst in devices[dclass]:
                     for gpuid in gpus:
-                        bus_id = devices[cls][inst]['bus_id'].lower()
-                        if bus_id == gpus[gpuid]:
-                            devices['gpu'][gpuid] = devices[cls][inst]
+                        bus_id = devices[dclass][inst]['bus_id'].lower()
+                        if bus_id and bus_id == gpus[gpuid]:
```

Comment:
I'm not sure you need the check 'if bus_id'.  You're calling bus_id.lower() the line above.  That will likely fail otherwise.  Maybe you want to put that if above?